File: Tienda_Pyton/Sistema.py

from Tienda import Tienda
class Sistema:
    CONTRASENA_SEGURA = "H1/m"
    tienda = Tienda()
    
    def ejecutar_programa(self):
        contrasena_valida = False
        while not contrasena_valida:
            print("***Bienvenido***")
            print("Ingresa la contraseña: ")
            contrasena = input()
            
            if self.validar_contrasena(contrasena):
                contrasena_valida = True
                # Mostrar menu
                self.mostrar_menu_sistema()
            else:
                print("Contrasena invalida, intenta de nuevo")
    
    def validar_contrasena(self, contrasena):
        return contrasena == self.CONTRASENA_SEGURA
    
    def mostrar_menu_sistema(self):
        opcion = 0
        while opcion != 13:
            print("***Bienvenido***")
            print("Elige una opcion para continuar")
            print("1. Agregar cliente")
            print("2. Agregar producto")
            print("3. Añadir stock")
            print("4. Disminuir stock")
            print("5. Consultar clientes")
            print("6. Consultar productos")
            print("7. Consultar productos por categoria")
            print("8. Realizar una compra")
            print("9. Ver compras")
            print("10. Consultar total de clientes y productos")
            print("11. Eliminar cliente")
            print("12. Eliminar producto")
            print("13. Salir")
            print("Ingresa la opcion: ")
            opcion = int(input())
            
            if opcion == 1:
                self.tienda.registrar_cliente()
            elif opcion == 2:
                print("Ingresa el tipo de producto: ")
                print("1. Limpieza")
                print("2. Electrodomesticos")
                print("3. Alimento")
                print("4. Maquillaje")
                opcion_producto = int(input())
                if opcion_producto == 1:
                    print("***Elegiste la opcion de registrar un producto de limpieza***")
                    self.tienda.registrar_producto_limpieza()
                elif opcion_producto == 2:
                    print("***Elegiste la opcion de registrar un producto de electrodomestico***")
                    self.tienda.registrar_producto_electrodomesticos()
                elif opcion_producto == 3:
                    print("***Elegiste la opcion de registrar un producto de alimento***")
                    self.tienda.registrar_producto_alimento()
                elif opcion_producto == 4:
                    print("***Elegiste la opcion de registrar un producto de maquillaje***")
                    self.tienda.registrar_producto_maquillaje()
            elif opcion == 5:
                self.tienda.consultar_clientes()
            elif opcion == 6:
                self.tienda.consultar_productos()
            elif opcion == 7:
                print("Elegiste la opcion ver productos por categoria: ")
                print("Ingresa el tipo de producto: ")
                print("1. Limpieza")
                print("2. Electrodomesticos")
                print("3. Alimento")
                print("4. Maquillaje")
                opcion_ver = int(input())
                if opcion_ver == 1:
                    print("***Elegiste la opcion de ver productos de limpieza***")
                    self.tienda.registrar_producto_limpieza()
                elif opcion_ver == 2:
                    print("***Elegiste la opcion de ver productos de electrodomestico***")
                    self.tienda.registrar_producto_electrodomesticos()
                elif opcion_ver == 3:
                    print("***Elegiste la opcion de ver productos de alimento***")
                    self.tienda.registrar_producto_alimento()
                elif opcion_ver == 4:
                    print("***Elegiste la opcion de ver productos de maquillaje***")
                    self.tienda.registrar_producto_maquillaje()
            elif opcion == 8: 
                    print("***Elegiste la opcion de realizar una compra***")
                    self.tienda.comprar_productos()
            elif opcion == 9: 
                    print("***Elegiste la opcion de ver todas las compras***")
                    self.tienda.ver_compras()
            elif opcion == 10: 
                    print("***Elegiste la opcion de ver total de productos y clientes***")
                    self.tienda.consultar_total_de_clientes_y_productos()
            elif opcion == 11: 
                    print("***Elegiste la opcion de eliminar clientes***")
                    self.tienda.eliminar_cliente()
            elif opcion == 12: 
                    print("***Elegiste la opcion de eliminar producto***")
                    self.tienda.eliminar_producto()
            elif opcion == 13: 
                    print("Saliendo...")
    

File: Tienda_JAVA/src/javaapplication45/Sistema.java

package javaapplication45;
import java.util.*;
public class Sistema {
    private final String CONTRASENA_SEGURA = "H1/m";
    Tienda tienda = new Tienda ();
    Scanner leer = new Scanner (System.in);
    
    public void ejecutarPrograma(){
        boolean contrasenaValida = false;
        do{
            System.out.println("***Bienvenido***");
        System.out.println("Ingresa la contraseña: ");
         String contrasena = leer.nextLine();
        
        if(validarContrasena(contrasena)){
            contrasenaValida = true;
            //Mostrar menu
            this.mostrarMenuSistema();
        }
        else {
            System.out.println("Contrasena invalida, intenta de nuevo");
        }
        }while(!contrasenaValida);
        
    }
   
    private boolean validarContrasena(String contrasena){
        return contrasena.equals(CONTRASENA_SEGURA);
    }
    
    private void mostrarMenuSistema(){
        int opcion = 0;
        do{
            System.out.println("***Bienvenido***");
            System.out.println("Elige una opcion para continuar");
            System.out.println("1. Agregar cliente");
            System.out.println("2. Agregar producto");
            System.out.println("3. Añadir stock");
            System.out.println("4. Disminuir stock");
            System.out.println("5. Consultar clientes");
            System.out.println("6. Consultar productos");
            System.out.println("7. Consultar productos por categoria");
            System.out.println("8. Realizar una compra");
            System.out.println("9. Ver compras");
            System.out.println("10. Consultar total de clientes y productos");
            System.out.println("11. Eliminar cliente");
            System.out.println("12. Eliminar producto");
            System.out.println("13. Salir");
            System.out.print("Ingresa la opcion: ");
            opcion = leer.nextInt();
            
            switch ( opcion ){
                case 1: tienda.registrarCliente();
                break;
                    
                case 2: 
                    System.out.println("Ingresa el tipo de producto: ");
                    System.out.println("1. Limpieza");
                    System.out.println("2.Electrodomesticos");
                    System.out.println("3.Alimento");
                    System.out.println("4.Maquillaje");
                    int opcionProducto = leer.nextInt();
                    switch(opcionProducto){
                        case 1:
                            System.out.println("***Elegiste la opcion de registrar un producto de limpieza***");
                            tienda.registrarProductoLimpieza();
                        break;
                        case 2:
                            System.out.println("***Elegiste la opcion de registrar un producto de electrodomestico***");
                            tienda.registrarProductoElectrodomesticos();
                        break;
                        case 3:
                            System.out.println("***Elegiste la opcion de registrar un producto de alimento***");
                            tienda.registrarProductoAlimento();
                        break;
                        case 4:
                            System.out.println("***Elegiste la opcion de registrar un producto de maquillaje***");
                            tienda.registrarProductoMaquillaje();
                        break;
                        
                    }
                break;
                case 5: tienda.consultarClientes();
                    break;
                case 6: tienda.consultarProductos();
                    break;
                    
                case 7:
                    System.out.println("Elegiste la opcion ver productos por categoria: ");
                    System.out.println("Ingresa el tipo de producto: ");
                    System.out.println("1. Limpieza");
                    System.out.println("2.Electrodomesticos");
                    System.out.println("3.Alimento");
                    System.out.println("4.Maquillaje");
                    int opcionver=leer.nextInt();
                    switch(opcionver){
                        case 1:
                            System.out.println("***Elegiste la opcion de ver productos de limpieza***");
                            tienda.registrarProductoLimpieza();
                        break;
                        case 2:
                            System.out.println("***Elegiste la opcion de ver productos de electrodomestico***");
                            tienda.registrarProductoElectrodomesticos();
                        break;
                        case 3:
                            System.out.println("***Elegiste la opcion de ver productos de alimento***");
                            tienda.registrarProductoAlimento();
                        break;
                        case 4:
                            System.out.println("***Elegiste la opcion de ver productos de maquillaje***");
                            tienda.registrarProductoMaquillaje();
                        break;
                    }
                    break;
                case 8: 
                    System.out.println("***Elegiste la opcion de realizar una compra***");
                    tienda.comprarProductos();
                    break;
                case 9: 
                    System.out.println("***Elegiste la opcion de ver todas las compras***");
                    tienda.verCompras();
                    break;
                case 10: 
                    System.out.println("***Elegiste la opcion de ver total de productos y clientes***");
                    tienda.consultarTotalDeClientesYProductos();
                    break;
               case 11: 
                    System.out.println("***Elegiste la opcion de eliminar clientes***");
                    tienda.eliminarCliente();
                    break;
                case 12: 
                    System.out.println("***Elegiste la opcion de eliminar producto***");
                    tienda.eliminarProducto();
                    break;
                case 13: 
                    System.out.println("Saliendo...");
                    break;
            }
        }while(opcion !=13);
        
        
    }
}
